src/strategy/strategy.ts

```diff
@@ -215,4 +218,31 @@ export class Strategy {
   ) {
     return this.checkConstraints(context, constraints) && this.isEnabled(parameters, context);
   }
+
+  getResult(parameters: any,
```

Comment:
this is the main change - everything strategy variant related is located in one class